data/shell-completions/zsh/meson-comp-helper.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/python3
+
+'''This file generates completion function definitions for Zsh.'''
+# NB: the function definitions are directly passed to the `eval` builtin.
+# We must take care not to trust meson introspect's output
+# and prevent e. g. code injection.
+
+import json
+import os
+import sys
+from shlex import quote
+
+
+def fsemic(s):
+    return s.replace(':', r'\:').replace(r'\\:', r'\:')
+
+
+def dsemic(s):
+    return s.replace(':', r'\\:')
+
+
+def c_style_octal(i):
+    return '0{:03}'.format(int(oct(i)[2:]))
+
+
+def generate_choices(optentry):
+    '''If a build option can be set to a limited number of values,
+       return an iterable of possible values. Otherwise, return None.
+    '''
+    if optentry['type'] in ('combo', ):
+        return optentry.get('choices', [])
+    if optentry['type'] in ('boolean', ):
+        return ('false', 'true')
+    return None
+
+
+def simple_hint(i):
+    return '<{}>'.format(i)
+
+
+def generate_hint(optentry):
+    '''Produce a string representation of a build option's current value.
+    '''
+    v = optentry['value']
+
+    if optentry['type'] in ('boolean', ):
+        return simple_hint('true' if v else 'false')
+    if optentry['name'] == 'install_umask':
+        return simple_hint(c_style_octal(v))
+    if optentry['type'] in ('integer', ):
```

Comment:
You're mixing `if x in (y, )` and `if x == y`. It's a bit strange to se if `x in single element inline container`, so maybe just `if x == y`

Reply:
The `if x in (y,)` form for the type matches is there in case there are new option types in the future that behave the same way; shorter diffs, etc. I've taken another look at it, though; I now believe that is unlikely to ever happen.

Will fix, thanks.